packages/native-stack/src/types.tsx

```diff
@@ -452,6 +454,43 @@ export type NativeStackNavigationOptions = {
    * @platform ios
    */
   gestureResponseDistance?: ScreenProps['gestureResponseDistance'];
+  /**
+   * Changes the gesture for dismissing the screen to the previous one.
+   *
+   * Supported values:
+   * - "swipeRight": swipe right to dismiss the screen
+   * - "swipeLeft": swipe left to dismiss the screen
+   * - "swipeUp": swipe up to dismiss the screen
+   * - "swipeDown": swipe down to dismiss the screen
+   * - "verticalSwipe": swipe up or down to dismiss the screen
+   * - "horizontalSwipe": swipe left or right to dismiss the screen
+   * - "twoDimensionalSwipe": swipe in any direction to dismiss the screen
+   *
+   * Only supported on iOS and Android.
+   */
+  goBackGesture?: GoBackGesture;
+  /**
+   * Changes the animation for dismissing the screen when making a swipe gesture.
+   * You can choose one of the built-in transition presets, by using `ScreenTransition` from `react-native-reanimated`.
+   *
+   * Supported values:
+   * - ScreenTransition.SwipeRight
+   * - ScreenTransition.SwipeLeft
+   * - ScreenTransition.SwipeUp
+   * - ScreenTransition.SwipeDown
+   * - ScreenTransition.Horizontal
+   * - ScreenTransition.Vertical
+   * - ScreenTransition.TwoDimensional
+   * - ScreenTransition.SwipeRightFade
+   *
+   * You can also construct custom screen transition by providing an object with `topScreenFrame` and `belowTopScreenFrame` properties.
+   * Only supported on iOS and Android.
+   */
+  transitionAnimation?: AnimatedScreenTransition;
+  /**
+   * Whether the screen should allow for making a swipe gesture from screen edges. Defaults to `true`.
+   */
+  screenEdgeGesture?: boolean;
```

Comment:
I think something like `screenEdgeGestureEnabled` would be better following the convention for boolean options that we normally use.

Or maybe `gestureFromEdgeEnabled` to also follow the convention of starting gesture related options with `gesture` prefix.

Reply:
Done in https://github.com/react-navigation/react-navigation/pull/11943/commits/301cd8dace25c46ed8b0f5931333176b07dcc99c.